plugins/outputs/cloudwatchlogs/internal/pusher/target.go

```diff
@@ -54,6 +71,13 @@ func (m *targetManager) InitTarget(target Target) error {
 	return nil
 }
 
+func (m *targetManager) PutRetentionPolicy(target Target) {
+	if target.Retention > 0 {
```

Comment:
Just a note: We have deduping of log groups/retention policy as part of
https://github.com/aws/amazon-cloudwatch-agent/blob/799f0c3bdcfc18fcadf3b5e0c1c9ebaa543e18a8/logs/logs.go#L169-L178
where it sets duplicates to -1 and I believe there's validation to ensure that all retention values are the same for a log group in the configuration.